src/main/java/uk/org/tombolo/importer/phe/AdultObesityImporter.java

```diff
@@ -61,65 +57,165 @@ public DatasourceSpec getDatasourceSpec(String datasourceIdString) throws Except
-            ConstantExtractor attributeExtractor = new ConstantExtractor(attributeLabel.name());
-            RowCellExtractor valueExtractor
-                    = new RowCellExtractor(getAttributeColumnId(attributeLabel), CellType.NUMERIC);
-            timedValueExtractors.add(new TimedValueExtractor(
-                    getProvider(),
-                    subjectType,
-                    subjectExtractor,
-                    attributeExtractor,
-                    timestampExtractor,
-                    valueExtractor));
+
+        // We create an empty list that will keep our .csv values
+        List<TimedValue> timedValues = new ArrayList<TimedValue>();
+
+        CSVFormat format = CSVFormat.DEFAULT;
+        String fileLocation = getDatasourceSpec("adultObesity").getUrl();
+
+        // The code below fetches the .csv file from the URL we specified in our DatasourceSpec object
+        URL url;
+        try {
+            url = new URL(fileLocation);
+        } catch (MalformedURLException e) {
+            File file;
+            if (!(file = new File(fileLocation)).exists()) {
+                log.error("ERROR: File does not exist: " + fileLocation);
+            }
+            url = file.toURI().toURL();
+        }
+
+        InputStreamReader isr = new InputStreamReader(
+                downloadUtils.fetchInputStream(url, getProvider().getLabel(), ".csv"));
+
+        // Parsing our csv file
+        CSVParser csvFileParser = new CSVParser(isr, format);
+        csvRecords = csvFileParser.getRecords();
+
+
+        // We discard the first 6 records in our data file as these don't hold any meaningfull information.
+        // We do this  calling an iterator object and simply ignoring them:
+        Iterator<CSVRecord> rowIterator = csvRecords.iterator();
+
+        // skipping first 6 rows
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+
+        // Looping through the rows of the .csv file
+        while (rowIterator.hasNext()){
+            CSVRecord row = rowIterator.next();
+            try{
+                Subject subject = SubjectUtils.getSubjectByTypeAndNameUnique(localauthority, String.valueOf(row.get(2)).trim());
+
+                // Checking not matched geometries
+                if (subject!=null){
+                    // Creating the time index
+                    for (int timeValuesIndex = 3; timeValuesIndex <= 5; timeValuesIndex++) {
+
+                        CSVRecord rowTime = (CSVRecord) csvRecords.get(5);
+                        String year = rowTime.get(timeValuesIndex).toString();
+
+                        // Cleaning the year record as it appears as: 2014 (Mid-January 2014 to Mid-January 2015)
+                        year = year.substring(0, 4);
+                        LocalDateTime timestamp = TimedValueUtils.parseTimestampString(year);
+
```

Comment:
above 4 lines can be moved out of every loop

Reply:
I'm not sure i understand...

Reply:
ignore my comment that wasn't right